packages/integrations/src/unifi-controller/unifi-controller-integration.ts

```diff
@@ -0,0 +1,274 @@
+        }
+
+        for (const health of site.health) {
+          switch(health.subsystem) {
+            case "wan":
+              if (health.status === "ok") {summary.wanStatus = "enabled"}
+              break;
+            
+            case "www":
+              if (health.status === "ok") {summary.wwwStatus = "enabled"}
+              summary.wwwLatency = health.latency;
+              summary.wwwPing    = health.speedtest_ping;
+              summary.wwwUptime  = health.uptime;
+              break;
+
+            case "wlan":
+              if (health.status === "ok") {summary.wifiStatus = "enabled"}
+              summary.wifiUsers  = health.num_user;
+              summary.wifiGuests = health.num_guest;
+              break;
+
+            case "lan":
+              if (health.status === "ok") {summary.lanStatus = "enabled"}
+              summary.lanUsers  = health.num_user;
+              summary.lanGuests = health.num_guest;
+              break;
+
+            case "vpn":
+              if (health.status === "ok") {summary.vpnStatus = "enabled"}
+              summary.vpnUsers  = health.remote_user_num_active;
+              break;
+
+            default:
+              break;
+          }
+        }
+      }
+
+    }
+
+    return summary;
+  }
+
+
+  public async testConnectionAsync(): Promise<void> {
+
+    await this.authenticateAndConstructSessionInHeaderAsync();
+
+  }
+
+  private headers: Record<string, string> | undefined = undefined;
+  private csrfToken: string | null | undefined;
+
+  
+  private async authenticateAndConstructSessionInHeaderAsync(): Promise<void> {
+    logger.debug("prefix:"+this.prefix); 
+
+    //let csrfToken: string | null | undefined;
+    if (!this.prefix) {
+      logger.debug("prefix not set; initial connect to determine UDM variant");
+      // check for UDM Pro and remember
+      const url = this.url("/");
+      logger.debug("url:"+url.toString());
+
+      const { status, ok, headers} = await fetchWithTrustedCertificatesAsync(url,  { method: 'HEAD', })
+      .then(res => res)
+      .catch((err: TypeError) => {
+        const detailMessage = String(err.cause);
+        throw new IntegrationTestConnectionError("invalidUrl", detailMessage);
+      });
+
+      logger.debug("status after initial connection: "+status); 
+
+      if (!ok) {
+        //Todo
+        throw new IntegrationTestConnectionError("invalidUrl", "status code: "+status);
+      }
+
+      let prefix = "";
+      if (headers.get("x-csrf-token") !== null) {
+        // Unifi OS < 3.2.5 passes & requires csrf-token
+        prefix = udmpPrefix;
+        this.csrfToken = headers.get("x-csrf-token");
+
+      } else if (headers.get("access-control-expose-headers") !== null) {
+        // Unifi OS ≥ 3.2.5 doesnt pass csrf token but still uses different endpoint
+        prefix = udmpPrefix;
+      }
+      this.prefix = prefix;
+      logger.debug("final prefix: "+this.prefix); 
+    }
+
+    logger.debug("trying log in");
```

Comment:
I will not further mention this over and over; please extend your logging statements so it is easier to understand where they were logged. It will help a lot when troubleshooting. Thank you!